File: src/decoders/assignment-theme.ts
import type { AssignmentTheme } from "~/models";
import { decodeSubject } from "./subject";

export const decodeAssignmentTheme = (theme: any): AssignmentTheme => ({
  id: theme.N,
  name: theme.L,
  subject: decodeSubject(theme.Matiere.V)
});

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/assignment-theme.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.AssignmentTheme
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive

fun decodeAssignmentTheme(theme: JsonObject) =
    AssignmentTheme(
        id = theme["N"]!!.jsonPrimitive.content,
        name = theme["L"]!!.jsonPrimitive.content,
        subject = decodeSubject(theme["Matiere"]!!.jsonObject["V"]!!.jsonObject))
